pkg/trace/agent/agent.go

```diff
@@ -349,6 +355,23 @@ func newChunksArray(chunks []*pb.TraceChunk) []*pb.TraceChunk {
 
 var _ api.StatsProcessor = (*Agent)(nil)
 
+// filterSpans removes all spans for which the provided FilterSpan function returns true
+func (a *Agent) filterSpans(p *api.Payload) {
+	if a.FilterSpan == nil {
+		return
+	}
+	for i := 0; i < len(p.Chunks()); i++ {
+		chunk := p.Chunk(i)
+		filteredSpans := []*pb.Span{}
```

Comment:
I'm not sure what the general use case for this is, but it might be more efficient to pre-allocate for the size:
```suggestion
		out := make([]*pb.Span, 0, len(chunk.Spans))
```
I assume many times no spans are filtered, or that very few are. But you should know this use case better.

Lastly, please use a shorter name for this variable.

Reply:
> Your suggestion is good to avoid allocating unnecessarily but in the case that we filter out a large number of spans we'd be allocating a bunch of unused space. Do you think this is a good trade off?

I will let you decide. I just wanted to raise the thought.

> Also, I think the name filteredSpans (or removedSpans, deletedSpans, discardedSpans or anything else) is relatively descriptive. Do you have any suggestions? I'm not sure how short it should be.

I agree that they are very descriptive. That's part of the issue. My suggestion was calling it `out`. I think that we are already within the context of a function called `filterSpans`, and within that context, a variable called `out` is pretty self explanatory.

Reply:
Thanks! Resolving!